src/runtime/managedtype.cs

```diff
@@ -80,48 +79,41 @@ internal void FreeGCHandle()
         /// </summary>
         internal static ManagedType GetManagedObject(IntPtr ob)
         {
-            if (ob != IntPtr.Zero)
+            if (ob == IntPtr.Zero)
             {
-                IntPtr tp = Runtime.PyObject_TYPE(ob);
-                if (tp == Runtime.PyTypeType || tp == Runtime.PyCLRMetaType)
-                {
-                    tp = ob;
-                }
-
-                var flags = Util.ReadCLong(tp, TypeOffset.tp_flags);
-                if ((flags & TypeFlags.Managed) != 0)
-                {
-                    IntPtr op = tp == ob
-                        ? Marshal.ReadIntPtr(tp, TypeOffset.magic())
-                        : Marshal.ReadIntPtr(ob, ObjectOffset.magic(tp));
-                    if (op == IntPtr.Zero)
-                    {
-                        return null;
-                    }
-                    var gc = (GCHandle)op;
-                    return (ManagedType)gc.Target;
-                }
+                return null;
+            }
+            IntPtr tp = Runtime.PyObject_TYPE(ob);
+            var flags = Util.ReadCLong(tp, TypeOffset.tp_flags);
+            if ((flags & TypeFlags.Managed) == 0)
+            {
+                return null;
+            }
+            int offset = ObjectOffset.magic(tp);
+            if (offset == 0)
```

Comment:
A managed wrapped type which released the corresponded GCHandle, that would be 0.

Reply:
But `magic` is the offset of the handle, not the handle itself.

Reply:
Ah, my bad, it should represent a managed type without GCHandle mapped, like `CLR Metatype`. You can think the new module of extension contains such a slot, but it might be 0.

Reply:
> it might be 0

what is the scenario for that?